clang/lib/Driver/ToolChains/Arch/LoongArch.cpp

```diff
@@ -135,8 +135,13 @@ void loongarch::getLoongArchTargetFeatures(const Driver &D,
     Features.push_back("+lsx");
 
   std::string ArchName;
-  if (const Arg *A = Args.getLastArg(options::OPT_march_EQ))
+  if (const Arg *A = Args.getLastArg(options::OPT_march_EQ)) {
     ArchName = A->getValue();
+    if (ArchName == "native")
+      for (auto &F : llvm::sys::getHostCPUFeatures())
+        Features.push_back(
```

Comment:
It's better to push these features after `llvm::LoongArch::getArchFeatures(ArchName, Feature)` is called. Because `getHostCPUFeatures()` is more exact than the latter.

Reply:
ArchName `"native"` will be lost after `postProcessTargetCPUString(ArchName, Triple);`

It needs to make a copy of `"native"` or use the original `A->getValue()`  after `getHostCPUFeatures`

Reply:
It changes the behavior that `-march` is not specified.

How about:
```
  std::string ArchName;
  const Arg *MArch = Args.getLastArg(options::OPT_march_EQ);
  if (MArch)
    ArchName = A->getValue();
  ArchName = postProcessTargetCPUString(ArchName, Triple);
  llvm::LoongArch::getArchFeatures(ArchName, Features);
  if (MArch->getValue() == "native")
    for(...) push_back(xxx);
```